src/sentry/api/serializers/rest_framework/notification_action.py

```diff
@@ -262,6 +262,10 @@ def validate(self, data: NotificationActionInputData) -> NotificationActionInput
         data = self.validate_slack_channel(data)
         data = self.validate_pagerduty_service(data)
 
+        # we may pass in a list of projects in which the user has project membership
+        if self.context.get("projects"):
+            data["projects"] = [project.id for project in self.context["projects"]]
```

Comment:
this is a little funky. i'm not sure if it would be better to pass a copy of a mutated `request.data` / pass extra context into the serializer with the correct list of projects that the user actually has access to, or to return `PermissionDenied` because the user doesn't have access to all of the projects in which they are trying to create a notification action for.

i went with the former here

Reply:
i'm not aware of us having any conventions for this, although i'm now thinking it should fail if you don't have access to all of them, like you said. this makes the logic simpler too